packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx

```diff
@@ -51,11 +51,27 @@ function NavbarContentLayout({
 
 export default function NavbarContent(): JSX.Element {
   const mobileSidebar = useNavbarMobileSidebar();
+  const windowSize = useWindowSize();
+  const isMobile = windowSize === 'mobile';
 
   const items = useNavbarItems();
   const [leftItems, rightItems] = splitNavbarItems(items);
+  const isSearchItem = (item: NavbarItemConfig) => item.type === 'search';
+  const hasExplicitSearchItem = items.some(isSearchItem);
 
-  const autoAddSearchBar = !items.some((item) => item.type === 'search');
+  const rightMostItems = [
+    <NavbarColorModeToggle
+      key="toggle"
+      className={isMobile ? 'margin-left--sm' : undefined}
+    />,
+    !hasExplicitSearchItem || isMobile ? <SearchBar /> : null,
+  ];
+  if (isMobile) {
+    [rightMostItems[0], rightMostItems[1]] = [
+      rightMostItems[1]!,
+      rightMostItems[0]!,
+    ];
+  }
```

Comment:
Since the search button just opens a modal, can we not do the swap and always put the search as the leftmost item?

Reply:
I did this intentionally because [some custom search input](https://github.com/praveenn77/docusaurus-lunr-search) can be expanded by click, so it's _probably_ better to try to freeze the color mode toggle position. I may be wrong though, and it's a completely normal option, just compare:

| Option 1 (default)   |  Option 2 (with swapping items on mobiles)    |
| -------- | -------- |
| ![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/4408379/159856203-e460ef53-94a7-4953-a8d8-7aefe765548b.gif) | ![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/4408379/159855849-d3d4e00f-5274-4444-80a7-4e29117a725e.gif) |

Reply:
I see. If there are custom search inputs that don't open a modal then it makes sense

Reply:
Not so fan of this change in terms of UX and SSR/CSR behavior

This swap of theme/search leads to an issue. 

On a mobile with slow connexion:
- you see first theme/search
- it then flips to search/theme